specs/corla.pvs

```diff
@@ -0,0 +1,923 @@
+  issue_credential: TYPE =
+    [cdos, state_election, person -> [person, two_factor_authentication]]
+
+%%% What are the credentials that have been issued to this person?
+  credential?: [person -> two_factor_authentication]
+
+%%% Is this person authenticated?
+  authenticated: [person, two_factor_authentication -> bool]
+
+%%% What follows are several features used to authenticate various
+%%% roles to their respective dashboards.
+
+%%% Each dashboard has an abstract state machine that captures the
+%%% dashboard's workflow and consequntly identifies which features are
+%%% visible to its users at various stages. Those abstract state
+%%% machines and features are modeled in the dashboard modules
+%%% below. We focus only on authentication here.
+
+% @todo kiniry Introduce an authentication monad and a state monad for
+% our ASMs.
+
+% @description Authenticate this county administrator to that county dashboard.
+  authenticate_county_administrator:
+    [county_dashboard, county_administrator, two_factor_authentication -> bool]
+% @description Authenticate this state administrator to that state-wide dashboard.
+  authenticate_state_administrator:
+    [state_wide_dashboard, state_administrator, two_factor_authentication -> bool]
+END authentication
+
+%%% The user interfaces of the system are the visible interactive
+%%% parts of an application. There are several different user
+%%% interfaces in the RLA Tool. The precise names of these interfaces
+%%% is still under discussion, but basically one is for CDOS personnel
+%%% responsible for the audit at the state-level, one is for county
+%%% personnel at the county-level, and one is for audit board members.
+
+user_interface: DATATYPE
+BEGIN
+  uploading_interface?: uploading_interface
+  cvr_uploading_interface?: cvr_uploading_interface
+  county_auditing_interface?: county_auditing_interface
+  audit_adjudication_interface?: audit_adjudication_interface
+END user_interface
+
+%%% After a county administrator attempts to upload artifacts to the
+%%% RLA system's server, one of several different messages is shown.
+%%% Each is self-explanatory in this domain model, and each must be
+%%% used in a scenario of the system.
+
+% @relation ok inherit upload_system_message
+% @relation hash_wrong inherit upload_system_message
+% @relation file_type_wrong inherit upload_system_message
+% @relation data_transmission_interrupted inherit upload_system_message
+% @relation too_late inherit upload_system_message
+upload_system_message: DATATYPE
+  WITH SUBTYPES upload_successful, hash_wrong, file_type_wrong,
+                data_transmission_interrupted, too_late
+BEGIN
+  upload_successful?: make_upload_successful: upload_successful
+  hash_wrong?: make_hash_wrong: hash_wrong
+  file_type_wrong?: make_file_type_wrong: file_type_wrong
+  data_transmission_interrupted?: make_data_transmission_interrupted:
+    data_transmission_interrupted
+  too_late?: make_too_late: too_late
+END upload_system_message
```

Comment:
Add: `no voter-verifiable paper record found`, for situations where the tabulated ballot was not voter-verifiable, or is a duplicated ballot, not the original VVPR, and the original could not be located, or the original is a non-voter-verifiable ballot.